### ISSUE TITLE
Missing support for nested NOT/AND/OR rule statements with WAFv2

### ISSUE THREAD

**Issue by PeterBurner** [terraform-providers/terraform-provider-aws#13862]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.12.26
+ provider.archive v1.3.0
+ provider.aws v2.67.0
+ provider.null v2.1.2
+ provider.random v2.2.1
```

### Affected Resource(s)

* aws_wafv2_web_acl

### Terraform Configuration Files

```hcl
resource "aws_wafv2_web_acl" "waf" {
  name        = "waf"
  description = "WAF"
  scope       = "REGIONAL"

  default_action {
    allow {}
  }

  rule {
    name     = "rule"
    priority = 0

    action {
      block {}
    }

    statement {
      rate_based_statement {
        limit              = 300
        aggregate_key_type = "IP"

        scope_down_statement {
          not_statement {
            statement {
              or_statement {
                statement {
                  regex_pattern_set_reference_statement {
                    arn = aws_wafv2_regex_pattern_set.ignored_paths.arn
                    field_to_match {
                      uri_path {}
                    }
                    text_transformation {
                      type     = "LOWERCASE"
                      priority = 1
                    }
                  }
                }
                statement {
                  ip_set_reference_statement {
                    arn = aws_wafv2_ip_set.ignored_ips.arn
                  }
                }
              }
            }
          }
        }
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = false
      metric_name                = "rule"
      sampled_requests_enabled   = false
    }
  }

  visibility_config {
    cloudwatch_metrics_enabled = false
    metric_name                = "waf"
    sampled_requests_enabled   = false
  }
}
```

### AWS WAFv2 rule JSON  

```json
{
  "Name": "rule",
  "Priority": 0,
  "Statement": {
    "RateBasedStatement": {
      "Limit": 300,
      "AggregateKeyType": "IP",
      "ScopeDownStatement": {
        "NotStatement": {
          "Statement": {
            "OrStatement": {
              "Statements": [
                {
                  "IPSetReferenceStatement": {
                    "ARN": "arn:aws:wafv2:eu-central-1:XXX:regional/ipset/ignored-ips/XXX"
                  }
                },
                {
                  "RegexPatternSetReferenceStatement": {
                    "ARN": "arn:aws:wafv2:eu-central-1:XXX:regional/regexpatternset/ignored-paths/XXX",
                    "FieldToMatch": {
                      "UriPath": {}
                    },
                    "TextTransformations": [
                      {
                        "Priority": 0,
                        "Type": "LOWERCASE"
                      }
                    ]
                  }
                }
              ]
            }
          }
        }
      }
    }
  },
  "Action": {
    "Count": {}
  },
  "VisibilityConfig": {
    "SampledRequestsEnabled": false,
    "CloudWatchMetricsEnabled": false,
    "MetricName": "rule"
  }
}
```  

### Expected Behavior

Terraform deploys nested statements.

### Actual Behavior

`terraform apply` throws error:  
```
Error: Unsupported block type

  on waf/main.tf line 183, in resource "aws_wafv2_web_acl" "waf":
 183:               or_statement {

Blocks of type "or_statement" are not expected here.
```

When I create the rule with the JSON rule builder in the AWS Console it works.  
When I modify an existing rule which is managed by terraform in the AWS Console and use the rule JSON from above, `terraform apply` throws the following error:  
```
Error: Error setting rule: Invalid address to set: []string{"rule", "0", "statement", "0", "rate_based_statement", "0", "scope_down_statement", "0", "not_statement", "0", "statement", "0", "or_statement"}
```  
I tried doing this to check if my syntax was wrong. Terraform should have tried to revert the remote changes. The output of `terraform apply` could have helped me get the correct syntax. But it seems that Terraform actually does not support nested statements. It can't parse the remote changes.  

### Steps to Reproduce

1. deploy WAFv2 and working rule with Terraform  
2. try to use nested NOT/AND/OR statements in Terraform config --> Error  
3. revert statement changes in config  
4. modify working rule in AWS Console with rule JSON from above  
5. `terraform apply` --> Error

**Comment by anGie44** [comment]
hi @PeterBurner, thank you for reporting this issue! I was able to reproduce this as it looks like we're missing a level of recursion. I'm going to propose a fix in a forthcoming PR.

**Comment by anGie44** [comment]
The fix has been merged and will release with `v2.69.0` of the Terraform AWS Provider, expected in this week's release.
